ic-canister/ic-canister-macros/src/api.rs

```diff
@@ -346,7 +469,13 @@ fn get_args(sig: &Signature) -> Result<(Vec<Type>, Vec<Type>), Error> {
         ReturnType::Default => Vec::new(),
         ReturnType::Type(_, ty) => match ty.as_ref() {
             Type::Tuple(tuple) => tuple.elems.iter().cloned().collect(),
-            _ => vec![ty.as_ref().clone()],
+            ty => {
+                // Some types in trait canisters had to be marked as `AsyncReturn` as implementation detail
+                // but we do not need this when exporting them to candid files as ic calls them correctly
+                // in any case.
+                let extracted_type = crate::derive::extract_type_if_matches("AsyncReturn", ty);
+                vec![extracted_type.clone()]
```

Comment:
Can we support here both using the alias and the original type (`Box<Pin<...>>`)?

Reply:
Sure. Regarding docs it was planned, yes, after I deal with integrational tests in amm